Sources/Afluent/SequenceOperators/ThrottleSequence.swift

```diff
@@ -0,0 +1,165 @@
+                self.startInstant = startInstant
+            }
+            
+            func updateHasSeenFirstElement() {
+                lock.protect {
+                    hasSeenFirstElement = true
+                }
+            }
+            
+            func updateHasStartedInterval(_ hasStarted: Bool) {
+                lock.protect {
+                    hasStartedInterval = hasStarted
+                }
+            }
+            
+            func updateFirst(element: Element?) {
+                lock.protect {
+                    firstElement = element
+                }
+            }
+            
+            func updateLatest(element: Element?) {
+                lock.protect {
+                    latestElement = element
+                }
+            }
+            
+            func updateStart(instant: C.Instant) {
+                lock.protect {
+                    startInstant = instant
+                }
+            }
+        }
+        
+        public struct AsyncIterator: AsyncIteratorProtocol {
+            typealias Instant = C.Instant
+            
+            var upstream: Upstream
+            let interval: C.Duration
+            let clock: C
+            let latest: Bool
+            
+            var iterator: AsyncThrowingStream<(Element?, Element?), Error>.Iterator?
+            let intervalEvents = IntervalEvents()
+            
+            public mutating func next() async throws -> Element? {
+                try Task.checkCancellation()
+                
+                let clock = self.clock
+                let interval = self.interval
+                let upstream = self.upstream
+                let intervalEvents = self.intervalEvents
+                
+                if iterator == nil {
+                    self.iterator = AsyncThrowingStream<(Element?, Element?), Error> { continuation in
+                        
+                        let intervalTask = DeferredTask {
+                            if let intervalStartInstant = intervalEvents.startInstant {
+                                intervalEvents.updateHasStartedInterval(true)
+                                
+                                let intervalEndInstant = intervalStartInstant.advanced(by: interval)
+                                try await clock.sleep(until: intervalEndInstant, tolerance: nil)
+                                
+                                let firstElement = intervalEvents.firstElement
+                                let latestElement = intervalEvents.latestElement
+                                
+                                continuation.yield((firstElement, latestElement))
+                                
+                                intervalEvents.updateFirst(element: nil)
+                                intervalEvents.updateHasStartedInterval(false)
+                            }
+                        }
+                        
+                        Task {
```

Comment:
I know this is super annoying, but currently, since this is an unstrctured task, it won't get cancelled if the outer iterator is cancelled. Meaning the `for try await el in upstream` will continue to maintain it's "subscription", even if the outer is no longer "subscribed". Could you add cancellation of this task to the `onTermination` closure?